criu/pie/restorer.c

```diff
@@ -1724,7 +1758,11 @@ long __export_restore_task(struct task_restore_args *args)
 
 		while (nr) {
 			pr_debug("Preadv %lx:%d... (%d iovs)\n", (unsigned long)iovs->iov_base, (int)iovs->iov_len, nr);
-			r = sys_preadv(args->vma_ios_fd, iovs, nr, rio->off);
+			/*
+			 * If we're requested to punch holes in the file after reading we do
+			 * it to save memory. Limit the reads then to an arbitrary block size.
+			 */
+			r = preadv_limited(args->vma_ios_fd, iovs, nr, rio->off, args->auto_dedup ? 128 << 20 : 0);
```

Comment:
You need to define a constant for 128 << 20 and support it with a comment to explain its meaning and value.